Build failure with `coq-itree.5.2.0`

As seen in [this failing SAW CI job](https://github.com/GaloisInc/saw-script/actions/runs/8552771098/job/23435559610#step:9:36), `entree-specs` fails to build with the recently released `coq-itree.5.2.0`:

```
∗ installed coq-itree.5.2.0
[ERROR] The compilation of entree-specs.dev failed at "make".

#=== ERROR while compiling entree-specs.dev ===================================#
# context     2.1.5 | linux/x86_64 | ocaml-base-compiler.4.14.2 | pinned(git+https://github.com/GaloisInc/entree-specs.git#5cf91e69c08376bcb17a95a8d2bf2daf406ae8cd#5cf91e69c08376bcb17a95a8d2bf2daf406ae8cd)
# path        ~/work/saw-script/saw-script/_opam/.opam-switch/build/entree-specs.dev
# command     ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code   2
# env-file    ~/.opam/log/entree-specs-27938-01981c.env
# output-file ~/.opam/log/entree-specs-27938-01981c.out
### output ###
# coq_makefile -f _CoqProject -o Makefile.coq
# COQDEP VFILES
# *** Warning: in file theories/Eq/Eqit.v, library Basics.Tacs is required from root ITree and has not been found in the loadpath!
# *** Warning: in file theories/Ref/Padded.v, library Basics.Tacs is required from root ITree and has not been found in the loadpath!
# *** Warning: in file theories/Ref/EnTreeSpecDefinition.v, library Basics.Tacs is required from root ITree and has not been found in the loadpath!
# *** Warning: in file theories/Ref/MRecSpec.v, library Basics.Tacs is required from root ITree and has not been found in the loadpath!
# *** Warning: in file theories/Ref/EnTreeSpecFacts.v, library Basics.Tacs is required from root ITree and has not been found in the loadpath!
# *** Warning: in file theories/Ref/EnTreeSpecCombinatorFacts.v, library Basics.Tacs is required from root ITree and has not been found in the loadpath!
# *** Warning: in file theories/Ref/Concrete.v, library Basics.Tacs is required from root ITree and has not been found in the loadpath!
# COQC theories/Basics/HeterogeneousRelations.v
# COQC theories/Basics/QuantType.v
# COQC theories/Basics/PolyList.v
# COQC theories/Core/EnTreeDefinition.v
# COQC theories/Eq/Eqit.v
# File "./theories/Eq/Eqit.v", line 9, characters 0-130:
# Error: Cannot find a physical path bound to logical path
# Basics.Tacs with prefix ITree.
# 
# make[1]: *** [Makefile.coq:764: theories/Eq/Eqit.vo] Error 1
# make: *** [Makefile.coq:387: all] Error 2
```

It does, however, build with `coq-itree.5.1.2`, as seen in [this successful CI job](https://github.com/GaloisInc/saw-script/actions/runs/8520922439/job/23338965814#step:9:27).

Judging from the [`coq-itree` `CHANGELOG`](https://github.com/DeepSpec/InteractionTrees/blob/e7fed212b1061b358428b57e11ee489184e241a2/CHANGELOG.md#L6), this happens because `ITree.Basic.Tacs` was renamed to `ITree.Basic.Utils`. I'm unclear if there's a way to make Coq look for both directories.

For now, it's probably best just to impose an upper version bound of `coq-itree < 5.2` so that that version doesn't get picked in build plans.

--- a/opam
+++ b/opam
@@ -11,7 +11,7 @@ depends: [
   "ocaml"
   "ocamlfind"
   "coq" {>= "8.15"}
-  "coq-itree" {>= "5.0.0" & < "5.2"}
+  "coq-itree" {>= "5.2" & < "5.3"}
   "coq-paco" {>= "4.1.2"}
 ]
 build: [

--- a/opam
+++ b/opam
@@ -11,7 +11,7 @@ depends: [
   "ocaml"
   "ocamlfind"
   "coq" {>= "8.15"}
-  "coq-itree" {>= "5.0.0"}
+  "coq-itree" {>= "5.0.0" & < "5.2"}
   "coq-paco" {>= "4.1.2"}
 ]
 build: [

--- a/theories/Eq/Eqit.v
+++ b/theories/Eq/Eqit.v
@@ -8,7 +8,7 @@ From Paco Require Import paco.
 
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
      Eq.Paco2
      Basics.Monad

--- a/theories/Ref/AddSawCalls.v
+++ b/theories/Ref/AddSawCalls.v
@@ -1,6 +1,6 @@
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
  .
 From EnTree Require Import

--- a/theories/Ref/Concrete.v
+++ b/theories/Ref/Concrete.v
@@ -8,7 +8,7 @@ From Paco Require Import paco.
 
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
      Basics.Monad
      Eq.Paco2.

--- a/theories/Ref/EnTreeSpecCombinatorFacts.v
+++ b/theories/Ref/EnTreeSpecCombinatorFacts.v
@@ -6,7 +6,7 @@ From Coq Require Import
 
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
  .
 From EnTree Require Import

--- a/theories/Ref/EnTreeSpecDefinition.v
+++ b/theories/Ref/EnTreeSpecDefinition.v
@@ -8,7 +8,7 @@ From Paco Require Import paco.
 
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
      Basics.Monad
      Eq.Paco2.

--- a/theories/Ref/EnTreeSpecFacts.v
+++ b/theories/Ref/EnTreeSpecFacts.v
@@ -6,7 +6,7 @@ From Coq Require Import
 
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
  .
 From EnTree Require Import

--- a/theories/Ref/Example.v
+++ b/theories/Ref/Example.v
@@ -1,6 +1,6 @@
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
  .
 From ITree Require Import Basics.Monad.

--- a/theories/Ref/MRecSpec.v
+++ b/theories/Ref/MRecSpec.v
@@ -8,7 +8,7 @@ From Coq Require Import
 
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
  .
 From EnTree Require Import

--- a/theories/Ref/Padded.v
+++ b/theories/Ref/Padded.v
@@ -8,7 +8,7 @@ From Coq Require Import
 
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
  .
 

--- a/theories/Ref/TpEnc.v
+++ b/theories/Ref/TpEnc.v
@@ -10,7 +10,7 @@ From Paco Require Import paco.
 
 From ITree Require Import
      Basics.Basics
-     Basics.Tacs
+     Basics.Utils
      Basics.HeterogeneousRelations
      Eq.Paco2.
 

